src/main/java/org/owasp/webgoat/container/MvcConfiguration.java

```diff
@@ -114,26 +154,37 @@ public AsciiDoctorTemplateResolver asciiDoctorTemplateResolver(Language language
+                                                        ITemplateResolver lessonThymeleafTemplateResolver) {
         SpringTemplateEngine engine = new SpringTemplateEngine();
         engine.setEnableSpringELCompiler(true);
         engine.addDialect(new SpringSecurityDialect());
         engine.setTemplateResolvers(
-                Set.of(lessonTemplateResolver, asciiDoctorTemplateResolver, springThymeleafTemplateResolver));
+                Set.of(lessonTemplateResolver, asciiDoctorTemplateResolver, lessonThymeleafTemplateResolver, springThymeleafTemplateResolver));
         return engine;
     }
 
     @Override
     public void addResourceHandlers(ResourceHandlerRegistry registry) {
-        registry.addResourceHandler("/images/**").addResourceLocations("classpath:/images/");
-        registry.addResourceHandler("/lesson_js/**").addResourceLocations("classpath:/js/");
-        registry.addResourceHandler("/lesson_css/**").addResourceLocations("classpath:/css/");
-        registry.addResourceHandler("/video/**").addResourceLocations("classpath:/video/");
+        //WebGoat internal
+        registry.addResourceHandler("/css/**").addResourceLocations("classpath:/webgoat/static/css/");
+        registry.addResourceHandler("/js/**")
+                .addResourceLocations("classpath:/webgoat/static/js/");
+        registry.addResourceHandler("/plugins/**").addResourceLocations("classpath:/webgoat/static/plugins/");
+        registry.addResourceHandler("/fonts/**").addResourceLocations("classpath:/webgoat/static/fonts/");
+
+        //WebGoat lessons
+        registry.addResourceHandler("/images/**").addResourceLocations(lessonScanner.applyPattern("classpath:/lessons/%s/images/").toArray(String[]::new));
+        registry.addResourceHandler("/lesson_js/**").addResourceLocations(lessonScanner.applyPattern("classpath:/lessons/%s/js/").toArray(String[]::new));
+        registry.addResourceHandler("/lesson_css/**").addResourceLocations(lessonScanner.applyPattern("classpath:/lessons/%s/css/").toArray(String[]::new));
+        registry.addResourceHandler("/lesson_templates/**").addResourceLocations(lessonScanner.applyPattern("classpath:/lessons/%s/templates/").toArray(String[]::new));
+        registry.addResourceHandler("/video/**").addResourceLocations(lessonScanner.applyPattern("classpath:/lessons/%s/video/").toArray(String[]::new));
     }
 
     @Bean
-    public PluginMessages pluginMessages(Messages messages, Language language) {
-        PluginMessages pluginMessages = new PluginMessages(messages, language);
+    public PluginMessages pluginMessages(Messages messages, Language language,
+                                         ResourcePatternResolver resourcePatternResolver) {
+        PluginMessages pluginMessages = new PluginMessages(messages, language, resourcePatternResolver);
         pluginMessages.setDefaultEncoding("UTF-8");
```

Comment:
I know that the following line has not been changed, but as we are refactoring, can we use here the already defined UTF8 constant  instead of duplicating its value.